phpunit/class-wp-theme-json-test.php

```diff
@@ -46,6 +46,47 @@ public static function set_up_before_class() {
 		static::$user_id = self::factory()->user->create();
 	}
 
+	// Pretty print CSS in test assertions so that it provides a better diff when a test fails.
+	// Without this: the failing test outputs the entire css string as being different.
+	// With this: the failing test only highlights the specific CSS rule that is different.
+	private static function pretty_print_css( $css ) {
+		$css = str_replace( '{', " {\n", $css );
+		$css = str_replace( '}', "}\n\n", $css );
+		$css = str_replace( ';', ";\n", $css );
+		return $css;
+	}
+
+	/**
+	 * Asserts that two CSS strings are equivalent, but outputs
+	 * pretty printed test failures along with the raw css.
+	 *
+	 * - The pretty printed assertions are useful for understanding what actually changed.
+	 * - The raw css is included as it's useful for updating the test case's expected CSS.
+	 *
+	 * Inspired by:
+	 * https://stackoverflow.com/questions/6832263/phpunit-assertion-failed-but-i-want-to-continue-testing
+	 *
+	 * @param string $expected The expected CSS.
+	 * @param string $actual   The actual CSS.
+	 */
+	private function assertSameCSS( $expected, $actual, $message = '' ) {
+		$pretty_expected = self::pretty_print_css( $expected );
+		$pretty_actual   = self::pretty_print_css( $actual );
```

Comment:
> but outputs pretty printed test failures along with the raw css

Just checking - is it just that we want to prettify the failure output? 

If so, does it make sense to only prettify things in the `catch` block?

Reply:
> Just checking - is it just that we want to prettify the failure output?

It's more that the `assertSame` method produces a better diff if it's passed css that has line breaks rather than one long string. The idea is that it'll just show the individual selectors/rules that have changed rather than the whole of the css stylesheet as it does in trunk. Background is I'm working on a PR (https://github.com/WordPress/gutenberg/pull/64076), but with the way it works in trunk it's really difficult to know exactly which selectors I've updated with the change. Especially if you're not as familiar with theme json like me. 😄 

I don't think it'd be possible to get the same level of output by doing the pretty printing in the catch block, unless a custom diff function is used, but it feels like overkill.

Reply:
Examples here:
- Before: https://github.com/WordPress/gutenberg/actions/runs/10156649825/job/28085502142?pr=64076
- After: https://github.com/WordPress/gutenberg/actions/runs/10174137732/job/28139475922?pr=64076

Reply:
Ah, got it. Thanks for the explanation - I didn't think about the change comparison. 

As you were!